tests/unit/amo/components/TestCategory.js

```diff
@@ -75,163 +81,229 @@ describe('Category', () => {
-        visibleAddonType: visibleAddonType(ADDON_TYPE_THEME),
-        slug: 'ad-block',
-      },
-    };
+    expect(root.find(CategoryHeader)).toHaveProp('category', fakeCategory);
   });
 
-  beforeEach(() => {
-    store = dispatchClientMetadata().store;
-    store.dispatch(categoriesLoad({ result: [fakeCategory] }));
-  });
+  it('sets loading to true if categories are loading', () => {
+    _categoriesFetch();
+    const root = render({}, { autoDispatchCategories: false });
 
-  function _searchStart(props = {}) {
-    store.dispatch(searchStart({
-      errorHandlerId: 'Search',
-      ...props,
-    }));
-  }
+    expect(root.instance().props.loading).toEqual(true);
+  });
 
-  it('passes category filters', () => {
-    _searchStart({ filters });
-    const props = mapStateToProps(store.getState(), ownProps);
+  it('sets loading to false if nothing is loading', () => {
+    const root = render({}, { autoDispatchCategories: false });
 
-    expect(props).toEqual({
-      addonType: ADDON_TYPE_THEME,
-      category: null,
-      filters: { ...filters, page: 1 },
-      loading: true,
-      pathname: '/themes/ad-block/',
-      paginationQueryParams: { page: 1 },
-    });
+    expect(root.instance().props.loading).toEqual(false);
   });
 
-  it('gets category from filters/state/etc.', () => {
-    const filtersWithSlug = {
-      ...filters,
-      category: 'test',
+  describe('category lookup', () => {
+    const decoyCategory = {
+      ...fakeCategory,
+      application: CLIENT_APP_FIREFOX,
+      type: ADDON_TYPE_THEME,
+      slug: 'decoy-slug',
     };
-    const ownPropsWithSlug = {
-      location: { query: {} },
-      params: {
-        application: CLIENT_APP_FIREFOX,
-        visibleAddonType: visibleAddonType(ADDON_TYPE_THEME),
-        slug: 'test',
-      },
+    const targetCategory = {
+      ...fakeCategory,
+      application: CLIENT_APP_ANDROID,
+      type: ADDON_TYPE_EXTENSION,
+      slug: 'target-slug',
     };
-    _searchStart({ filters: filtersWithSlug });
-    const props = mapStateToProps(store.getState(), ownPropsWithSlug);
 
-    expect(props).toEqual({
-      addonType: ADDON_TYPE_THEME,
-      category: fakeCategory,
-      filters: { ...filtersWithSlug, page: 1 },
-      loading: true,
-      pathname: '/themes/test/',
-      paginationQueryParams: { page: 1 },
+    const _dispatchClientApp = (clientApp) => dispatchClientMetadata({
+      store, clientApp,
     });
-  });
 
-  it('sets loading to true if categories and search are loading', () => {
-    store.dispatch(_categoriesFetch());
-    _searchStart({ filters });
-    const props = mapStateToProps(store.getState(), ownProps);
+    function _render(
+      props = {}, { autoDispatchCategories = true, ...options } = {}
+    ) {
+      if (autoDispatchCategories) {
+        _categoriesLoad({ result: [decoyCategory, targetCategory] });
+      }
+      return render(props, {
+        // Since we loaded our own, tell the parent helper not to.
```

Comment:
"not to" seems an odd way to end a sentence. "Not to what?" was my first response reading the sentence.